rskj-core/src/main/java/co/rsk/peg/utils/BridgeEventLoggerImpl.java

```diff
@@ -75,10 +76,22 @@ public void logUpdateCollections(Transaction rskTx) {
     }
 
     @Override
-    public void logAddSignature(BtcECKey federatorPublicKey, BtcTransaction btcTx, byte[] rskTxHash) {
-        ECKey key = ECKey.fromPublicOnly(federatorPublicKey.getPubKey());
-        String federatorRskAddress = ByteUtil.toHexString(key.getAddress());
-        logAddSignatureInSolidityFormat(rskTxHash, federatorRskAddress, federatorPublicKey);
+    public void logAddSignature(FederationMember federationMember, BtcTransaction btcTx, byte[] rskTxHash) {
+        String federatorRskAddress = getFederatorRskAddress(federationMember);
+        logAddSignatureInSolidityFormat(rskTxHash, federatorRskAddress, federationMember.getBtcPublicKey());
+    }
+
+    private String getFederatorRskAddress(FederationMember federationMember){
+        if (shouldUseRskPublicKey()){
+            return ByteUtil.toHexString(federationMember.getRskPublicKey().getAddress());
+        }
+
+        ECKey ecKeyFromBtcPublicKey = ECKey.fromPublicOnly(federationMember.getBtcPublicKey().getPubKey());
+        return ByteUtil.toHexString(ecKeyFromBtcPublicKey.getAddress());
```

Comment:
```suggestion
       ECKey federatorPublicKey = getFederatorPublicKey();
       return ByteUtil.toHexString(federatorPublicKey.getAddress());


private EcKey getFederatorPublicKey() {
    if (shouldUseRskPublicKey()){
           return federationMember.getRskPublicKey();
    }else {
        return federationMember.getBtcPublicKey();
    }
```

Reply:
Done

Reply:
there is no need to use the `else` statement right? we are returning the value right after the if condition. I would do it this way:

```
private EcKey getFederatorPublicKey() {
    if (!shouldUseRskPublicKey()){
           return federationMember.getBtcPublicKey();
    }
    return federationMember.getRskPublicKey();
```

Reply:
Well, actually i'd do this:

```
private EcKey getFederatorPublicKey() {
    if (shouldUseBtcPublicKey()){
           return federationMember.getBtcPublicKey();
    }
    return federationMember.getRskPublicKey();
}

private boolean shouldUseBtcPublicKey() {
        return !activations.isActive(ConsensusRule.RSKIP415);
    }
```

Reply:
@marcos-iov @nathanieliov